packages/cli/src/services/user.service.ts

```diff
@@ -276,11 +283,83 @@ export class UserService {
 		return await this.userRepository.manager.transaction(async (trx) => {
 			await trx.update(User, { id: user.id }, { role: { slug: newRole.newRoleName } });
 
-			const adminDowngradedToMember =
-				user.role.slug === 'global:admin' && newRole.newRoleName === 'global:member';
+			const isAdminRole = (roleName: string) => {
+				return roleName === 'global:admin' || roleName === 'global:owner';
+			};
+
+			const isDowngradedToChatUser =
+				user.role.slug !== 'global:chatUser' && newRole.newRoleName === 'global:chatUser';
+			const isUpgradedChatUser =
+				user.role.slug === 'global:chatUser' && newRole.newRoleName !== 'global:chatUser';
+			const isDowngradedAdmin = isAdminRole(user.role.slug) && !isAdminRole(newRole.newRoleName);
+
+			if (isDowngradedToChatUser) {
+				// Revoke user's project roles in any shared projects they have access to.
+				const projectRelations = await trx.find(ProjectRelation, {
+					where: { userId: user.id, role: { slug: Not(PROJECT_OWNER_ROLE_SLUG) } },
+					relations: ['role'],
+				});
+				for (const relation of projectRelations) {
+					if (relation.role.slug === PROJECT_ADMIN_ROLE_SLUG) {
+						// Ensure there is at least one other admin in the project
+						const adminCount = await trx.count(ProjectRelation, {
+							where: {
+								projectId: relation.projectId,
+								role: { slug: In([PROJECT_ADMIN_ROLE_SLUG, PROJECT_OWNER_ROLE_SLUG]) },
+								userId: Not(user.id),
+							},
+						});
+						if (adminCount === 0) {
+							throw new UnexpectedError(
+								`Cannot downgrade user as they are the only project admin in project "${relation.projectId}".`,
+							);
```

Comment:
We don't want this reported to Sentry:

```suggestion
						if (adminCount === 0) {
							throw new UserError(
								`Cannot downgrade user as they are the only project admin in project "${relation.projectId}".`,
							);
```